recipe-app/ViewModel/RecipeListViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  RecipeListViewModel.swift
+//  recipe-app
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import Foundation
+import UIKit
+
+ class RecipeListViewModel{
+    let key = "5b91aa819e6d4bc8848f4c972103e6dc"
+    var recipes = [Recipe]()
+    
+     /*
+      Retrieves first 20 recipes matching the user's search.
+      */
+     func searchRecipe(_ query: String) {
```

Comment:
1. We can use a completion handler instead of console prints. You might want to notify other parts of your code with this response.
2. You only want to move to the main thread once you are updating your recipe data so there is no need to use `DispatchQueue.global` since you are already there. 
3. My suggestion is for the basic level. You might want to localize your network errors.
4. It's easier to read and safer to use `guard let` statements instead of highly coupled if `statements`. 

```suggestion
    func searchRecipe(_ query: String, 
                      completion: @escaping (Result<[Recipe], String>) -> Void) {
    guard let url = makeURL(query: query) else {
        completion(.failure("Invalid URL."))
        return
    }

    let session = URLSession.shared
    session.dataTask(with: url) { [weak self] data, response, error in
        guard let self = self else { return }

        if let error = error {
            completion(.failure("Network error occurred: \(error.localizedDescription)"))
            return
        }

        guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
            completion(.failure("Unexpected response."))
            return
        }

        guard let data = data else {
            completion(.failure("No data received from the server."))
            return
        }

        do {
            let recipes = try self.parse(json: data)
            DispatchQueue.main.async {
                self.recipes = recipes
                completion(.success(recipes))
            }
        } catch {
            completion(.failure("Failed to parse the response."))
        }
    }.resume()
}
```